### PR TITLE
Adding `--auto_rotate` argument

### PR DESCRIPTION
**Pull request by alexdesiqueira**
Mothra automatically tilting images could be cumbersome, mainly when EXIF data is not correct.
Here, we:
- Add the argument `--auto_rotate`; if it's used, Mothra will try to tilt images according to their EXIF orientation.
- Refactor the rotation functions through the pipeline and in `misc`.

### CODE REVIEW

File: pipeline.py [hunk 1]
```diff
@@ -72,6 +71,11 @@ def main():
                         action='store_true',
                         help='Enable computation cache (useful when developing algorithms)')
 
+    # Enable auto-rotation
+    parser.add_argument('--auto_rotate',
+                        action='store_true',
+                        help='Enable auto-rotation of input images')
```
Comment (stefanv) [inline pipeline.py:77]:
```suggestion
                        help='Enable rotation of input images based on EXIF tag')
```

File: mothra/misc.py [hunk 2]
```diff
@@ -27,6 +29,31 @@ def encodes(self, img: PILImage):
...
+    -------
+    image_rgb : 3D array
+        RGB image, rotated if angle in EXIF data is different than zero.
+    """
+    angle = read_angle(image_path)
+
+    print(f'Image angle: {angle} deg')
+
+    if angle not in (None, 0):  # angle == 0 does not need untilting
+        image_rgb = rotate(image_rgb, angle=angle, resize=True)
```
[the remark this answers is not in the document]
Comment (alexdesiqueira) [reply mothra/misc.py:52]: > Are you sure the angles are specified in the same direction?
According to EXIF documentation, the `exif` package, and our tests so far, that's appropriate. Could you clarify?

[the remark this answers is not in the document]
Comment (alexdesiqueira) [reply mothra/misc.py:52]: > Are you sure the angles are specified in the same direction?

According to EXIF documentation, the exif package, and our tests so far, that's appropriate. Could you clarify?

File: mothra/tests/test_misc.py [hunk 3]
```diff
@@ -1,11 +1,32 @@
 from glob import glob
 from mothra import misc
+from skimage.io import imread
 
 
 PATH_TEST_FILES = 'mothra/tests/test_files'
 TEST_INPUT_FILE = f'{PATH_TEST_FILES}/input_file.txt'
 TEST_INPUT_IMAGES = glob(f'{PATH_TEST_FILES}/test_input/*.JPG')
-TEST_TILTED_IMAGE =  f'{PATH_TEST_FILES}/test_input/BMNHE_1105737_17193_6eec94847b4939c6d117429d59829aac7a9fadf9.JPG'
+TEST_IMAGE_0DEG =  f'{PATH_TEST_FILES}/test_input/BMNHE_1105737_angle0.JPG'
```
[the remark this answers is not in the document]
Comment (alexdesiqueira) [reply mothra/tests/test_misc.py]: Replaced the images with smaller ones and used `exiftool` to edit their EXIF info.

**Comment by alexdesiqueira** [comment]
@stefanv on the slow processing for `test_pipeline.py`: it seems that it tries to run the pipeline in all images within the test folder. Maybe we could set one image only instead.

**Comment by alexdesiqueira** [comment]
@stefanv on the slow processing for `test_pipeline.py`: it seems that it tries to run the pipeline in all images within the test folder. We need some images more to test different conditions, now; I think they should all live at the same folder. Maybe we could set only one image to `test_pipeline` instead.

**Pull request merged**